analysis/filter/EdgeFilter/src/test/kotlin/de/maibornwolff/codecharta/filter/edgefilter/ProjectMergerTest.kt

```diff
@@ -88,59 +54,37 @@ class ProjectMergerTest : Spek({
-        }
+        assertEquals(project.sizeOfEdges(), 6)
+        assertEquals(leaf1.attributes.size, 2)
 
-        it("leaf1 should have correct number of attributes") {
-            MatcherAssert.assertThat(leaf1.attributes.size, CoreMatchers.`is`(2))
-        }
-
-        it("leaf1 should have correct pairingRate_relative value") {
-            val value: Int = getAttributeValue(leaf1.attributes, "pairingRate")
-            val expectedValue = (90 + 30 + 70) / 3 // see testfile
-            MatcherAssert.assertThat(value, CoreMatchers.`is`(expectedValue))
-        }
-
-        it("leaf1 should have correct avgCommits_absolute value") {
-            val value: Int = getAttributeValue(leaf1.attributes, "avgCommits")
-            val expectedValue = 30 + 10 + 30 // see testfile
-            MatcherAssert.assertThat(value, CoreMatchers.`is`(expectedValue))
-        }
-
-        it("leaf3 should have correct pairingRate_relative value") {
-            val number: Int = getAttributeValue(leaf3.attributes, "pairingRate")
-            val expectedValue = (90 + 60 + 70) / 3 // see testfile
-            MatcherAssert.assertThat(number, CoreMatchers.`is`(expectedValue))
-        }
-
-        it("leaf3 should have correct avgCommits_absolute value") {
-            val number: Int = getAttributeValue(leaf3.attributes, "avgCommits")
-            val expectedValue = 30 + 40 + 30 // see testfile
-            MatcherAssert.assertThat(number, CoreMatchers.`is`(expectedValue))
-        }
-
-        it("leaf4 should have correct pairingRate_relative value") {
-            val number: Int = getAttributeValue(leaf4.attributes, "pairingRate")
-            val expectedValue = (60 + 80 + 60) / 3 // see testfile
-            MatcherAssert.assertThat(number, CoreMatchers.`is`(expectedValue))
+        val leafs = listOf(leaf1, leaf3, leaf4)
+        // Check test files
+        val expectedPairingRates = listOf((90 + 30 + 70) / 3, (90 + 60 + 70) / 3, (60 + 80 + 60) / 3)
+        val expectedAvgCommits = listOf(30 + 10 + 30, 30 + 40 + 30, 20 + 30 + 40)
+        leafs.forEachIndexed { i, curLeaf ->
+            assertEquals(getAttributeValue(curLeaf.attributes, "pairingRate"), expectedPairingRates[i])
+            assertEquals(getAttributeValue(curLeaf.attributes, "avgCommits"), expectedAvgCommits[i])
         }
+    }
 
-        it("leaf4 should have correct avgCommits_absolute value") {
-            val number: Int = getAttributeValue(leaf4.attributes, "avgCommits")
-            val expectedValue = 20 + 30 + 40 // see testfile
-            MatcherAssert.assertThat(number, CoreMatchers.`is`(expectedValue))
-        }
+    @Test
+    fun `should contain attributeDescriptors after merge`() {
+        val originalProject = ProjectDeserializer.deserializeProject(
+            InputStreamReader(this.javaClass.classLoader.getResourceAsStream(TEST_EDGES_JSON_FILE_3)!!)
+        )
+        val project = EdgeProjectBuilder(originalProject, '/').merge()
```

Comment:
With what is the originalProject merged here? I think it would make sense to merge to different sets of attributeDescriptors from two different projects/jsons, right?

Reply:
The EdgeProjectBuilder does not have a build function, it only has the function called merge.
This is because it "merges" the edges into the existing nodes (if there are any). 
I will change the name of the test to clarify. This tests sole purpose is to verify, if the attributeDescriptor field gets copied while "merging"/building the new project out of the old parts.
I don't think that there is more to cover for testing attributeDescriptors in EdgeFilter, because if there were two sets of attributes, they were already combined in the attributeDescriptor-field before the filter becomes active, which means that a copy of the complete field is enough.
Or am I missing something here?

Reply:
Alright, that's fine for me.